In [1]:
import numpy as np
import pandas as pd
from scipy import stats

1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

s = 16
n = 256
m = 80


Найдем квантиль t-распределения
https://mipt-stats.gitlab.io/courses/python/07_scipy_stats.html
X(params).ppf(q) —  q -квантиль (Percent Point Function);

In [4]:
t_obj = stats.t(256 - 1)

In [6]:
t = t_obj.ppf(1 - 0.05 / 2)
t

1.9693105698491928

In [7]:
s = 16
n = 256
m = 80

In [9]:

fill_between=(m - t * s / np.sqrt(n), m + t * s / np.sqrt(n))

In [10]:
fill_between

(78.0306894301508, 81.9693105698492)

Ответ  78.0306894301508, 81.9693105698492

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

In [38]:
from statsmodels.stats.weightstats import _tconfint_generic as t_stat


https://www.statsmodels.org/dev/_modules/statsmodels/stats/weightstats.html

In [11]:
X = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
mean_X = X.mean()
std_X = X.std(ddof=1)
mean_std_X = std_X / (np.sqrt(len(X)))

In [23]:
std_X

0.4508017549014448

a/2 = 0,025
v = n-1 = 9
t = 2,262

In [42]:
t_stat(mean_X, mean_std_X,len(X) - 1, 0.05, 'two-sided')

(6.267515851415713, 6.912484148584288)

Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

H0=17mm, H1>17


In [51]:
Ho= (17.5 - 17) / 4 * np.sqrt(100)

In [52]:
Ho

1.25

F_ucr= (1-2*0.05)/2 в соответствии с таблицей = 1,65

F_ucr = ровна 1.65

Так как 1,25 > 1,645 то нулевую гипотезу можно принять, можно считать что
средний диаметр действительно d =17 мм. 

4.Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

H0: M = 200

H1: M != 200

In [60]:
X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
mean_X = X.mean()
std_X = X.std(ddof=1)

In [62]:
Ho = (mean_X - 200) / std_X * np.sqrt(10)
Ho


-1.0651074037450896

F_ucr =  (1-2*0.01)/2

In [65]:
F_ucr = 2,34

Ho>H1 нулевая гипотеза верна.